cos-gpu-installer-docker/entrypoint.sh

```diff
@@ -427,28 +437,63 @@ update_host_ld_cache() {
   ldconfig -r "${ROOT_MOUNT_DIR}"
 }
 
+usage() {
+  echo "usage: $0 [-p]"
+  echo "Default behavior installs all components needed for the Nvidia driver."
+  echo "  -p: Install cross toolchain package only"
+}
+
+parse_opt() {
+  while getopts ":ph" opt; do
+  case ${opt} in
+    p)
+      PRELOAD="true"
+      ;;
+    h)
+      usage
+      exit 0
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      usage
+      exit 1
+      ;;
+  esac
+  done
+}
+
 main() {
-  lock
-  load_etc_os_release
-  configure_kernel_module_locking
-  if check_cached_version; then
-    configure_cached_installation
-    verify_nvidia_installation
-    info "Found cached version, NOT building the drivers."
-  else
-    info "Did not find cached version, building the drivers..."
-    download_nvidia_installer
-    download_kernel_src
+  parse_opt "$@"
+  info "PRELOAD: ${PRELOAD}"
+  if [[ "$PRELOAD" == "true" ]]; then
+    load_etc_os_release
     set_compilation_env
     install_cross_toolchain_pkg
-    configure_nvidia_installation_dirs
-    configure_kernel_src
-    run_nvidia_installer
-    update_cached_version
-    verify_nvidia_installation
-    info "Finished installing the drivers."
+    download_kernel_src
```

Comment:
I think line 221-223 (`tar xf "${COS_KERNEL_SRC_ARCHIVE}"`) should be moved into the if block above to avoid duplicated decompression.

Reply:
Updated